### PR TITLE
[NEW] Support Multiple SNS Topic per agent/market

### PR DESCRIPTION
**Pull request by Shailesh351**
Part of https://github.com/WideChat/Rocket.Chat/issues/1226

### CODE REVIEW

**Comment by ear-dev** [comment]
@Shailesh351 did you mean to create this PR against the develop branch?  Shouldn't it be against develop_pwa at the moment?

File: app/livechat/server/lib/SNS.ts [hunk 1]
```diff
@@ -8,15 +8,23 @@ const logger = new Logger('SNS');
...
 	Object.assign(logData, data);
 	return logData;
 };
 
+const getSNSTopicARN = (roomId: string): string => {
+	const room = LivechatRooms.findOneLivechatById(roomId);
+	if (!room || !room.customFields?.sns) {
+		return '';
+	}
+	return room.customFields?.sns?.arn || '';
```
Comment (AlexanderKanakis) [inline app/livechat/server/lib/SNS.ts]:
```suggestion
	return room?.customFields?.sns?.arn || '';
```
Any reason not to simplify?

**Comment by AlexanderKanakis** [comment]
What is the format of the new `sns` customField?

File: app/livechat/server/lib/SNS.ts (hunk 1 above)
Comment (Shailesh351) [reply app/livechat/server/lib/SNS.ts]: Thanks Alexander!

**Comment by Shailesh351** [comment]
> What is the format of the new `sns` customField?


@AlexanderKanakis 
The payload format is the same as earlier. In this PR only the setting is moved from Server Omnichannel Settings to DF Config so that we can configure different SNS topics for different DF agents.

The way we save SNS data as the custom field is changed. 

```
sns: {
    arn: "<SNS ARN>"
    logData: {
        // all the data which we want to send to the SNS topic i.e. DF Agent ID
    }
}
```

When we want to store data that we don't like to send to SNS we store it directly as property of sns object i.e, `arn` 
This change is https://github.com/WideChat/Apps.Dialogflow/pull/169/files#diff-102ed9d697249891af92544b03600954a99bc60cdfd482da90a16b30abf6f13f

**Comment by Shailesh351** [comment]
> @Shailesh351 did you mean to create this PR against the develop branch? Shouldn't it be against develop_pwa at the moment?

Sorry @ear-dev, My bad! I have updated it to `develop_pwa`

**Review by AlexanderKanakis — APPROVED** [review]
I am unable to see the logs, but tested locally and logged sent events. Looks good.

**Pull request merged**